File: 2025-2/sesion-de-clase-02/Java/ejercicio01/src/main/java/pe/edu/pucp/inf30/ejercicio01/Profesor.java
package pe.edu.pucp.inf30.ejercicio01;

/**
 *
 * @author eric
 */
public class Profesor extends Persona {
    public Profesor() {
        System.out.println("Profesor: Constructor");
    }
    
    @Override
    public String toString() {
        return String.format("[Profesor] Nombre: %s, Edad: %d", this.getNombre(), this.getEdad());
    }
    
    @Override
    public void imprimir() {
        System.out.println(this);
    }
    
    Profesor(ProfesorBuilder builder) {
        super(builder);
    }
    
    public static class ProfesorBuilder extends Persona.PersonaBuilder {
        @Override
        public Profesor build() {
            return new Profesor(this);
        }
    }
}

File: 2025-2/sesion-de-clase-02/dotNET/Ejercicio01/Profesor.cs
using System;

namespace Ejercicio01 {
    public class Profesor : Persona {
        public Profesor() {
            Console.WriteLine("Profesor: Constructor");
        }

        Profesor(ProfesorBuilder builder) : base(builder) { 
        }

        public override string ToString() {
            return string.Format("[Profesor] Nombre: {0}, Edad: {1}", this.Nombre, this.Edad);
        }

        public override void Imprimir() {
            Console.WriteLine(this);
        }

        public class ProfesorBuilder : PersonaBuilder {
            public override Persona Build() {
                return new Profesor(this);
            }
        }
    }
}
